### ISSUE TITLE
[BUG] Some tests fail when building Data Prepper with non-US locale settings

### ISSUE THREAD

**Issue by kmssap** [opensearch-project/data-prepper#1270]
Some tests break when building Data Prepper with non-US locale settings. 

**Affected tests:**
- DefaultPluginFactoryIT (locale specific validation error)
- DateProcessorTests (locale specific date patterns)

**Comment by kmssap** [comment]
@chenqi0805 Yes those are the affected tests. Additionally `ApacheLogFakerTest#testRandomLogPattern` seems to break due to a locale-specific date pattern. I am happy to provide a PR

**Comment by dlvenable** [comment]
@kmssap , Yes, we'd be happy to review and merge a PR!
